### PR TITLE
Initial Implementation for hl7 connector

### PR DESCRIPTION
**Pull request by mayuravaani**
https://docs.google.com/document/d/1uiLAWphNBIytpo5rzSAdq-4pLxJH3XLvAf1uEvfJKFA/edit

## Automation tests
 - Unit tests 
   > Code coverage information : 89%

 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

**Comment by grainier** [comment]
@mayuravaani  Please fill in the PR template with a few details on what this PR about,
Purpose, Approach, User stories, Automation test, etc...

File: component/src/test/resources/security/kk.js [hunk 1]
```diff
@@ -0,0 +1 @@
+//Used to check invalid file.
```
Comment (AnuGayan) [inline component/src/test/resources/security/kk.js:1]: Why we need this file?

File: component/pom.xml [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            <artifactId>hapi-structures-v25</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>ca.uhn.hapi</groupId>
+            <artifactId>hapi-structures-v23</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.10</version>
```
Comment (AnuGayan) [inline component/pom.xml]: remove the version from here and define it in the parent pom. check the other places as well

File: component/pom.xml [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Name>${project.artifactId}</Bundle-Name>
+                        <Export-Package>
+                            org.wso2.extension.siddhi.io.hl7,
```
Comment (AnuGayan) [inline component/pom.xml]: provide the versions as well, https://github.com/wso2-extensions/siddhi-store-mongodb/blob/master/component/pom.xml#L322

File: component/pom.xml [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+                            org.wso2.extension.siddhi.io.hl7.*
+                        </Export-Package>
+                        <Import-Package>
+                            *;resolution:=optional
+                        </Import-Package>
+                        <Private-Package>
+                            ca.uhn.hl7v2.*
```
Comment (AnuGayan) [inline component/pom.xml]: provide the version.

File: component/pom.xml [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Name>${project.artifactId}</Bundle-Name>
+                        <Export-Package>
+                            org.wso2.extension.siddhi.io.hl7,
+                            org.wso2.extension.siddhi.io.hl7.*
+                        </Export-Package>
+                        <Import-Package>
+                            *;resolution:=optional
```
Comment (AnuGayan) [inline component/pom.xml]: can't we explicitly say what to import

File: component/pom.xml [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+                    </instructions>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.jacoco</groupId>
+                <artifactId>jacoco-maven-plugin</artifactId>
+            </plugin>
+        </plugins>
+    </build>
+</project>
```
Comment (AnuGayan) [inline component/pom.xml]: Add a new line at the end of the file

File: docs/images/siddhi-logo-w.svg [hunk 7]
```diff
@@ -0,0 +1,163 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!--
+  ~ Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (AnuGayan) [inline docs/images/siddhi-logo-w.svg]: Change this to 2019. Check other places as well

File: component/src/main/java/org/wso2/extension/siddhi/io/hl7/source/Hl7Source.java [hunk 8]
```diff
@@ -0,0 +1,308 @@
...
+
+        if (hl7Service != null) {
+            hl7Service.stop();
+        }
+
+    }
+
+    @Override
+    public void destroy() {
+
```
Comment (AnuGayan) [inline component/src/main/java/org/wso2/extension/siddhi/io/hl7/source/Hl7Source.java]: Shall we add a comment mentioning why this is empty? please do the same in other places as well

File: component/src/main/java/org/wso2/extension/siddhi/io/hl7/util/Hl7Utils.java [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+                throw new SiddhiAppCreationException("Failed to load keystore. please check the keystore defined in" +
+                        siddhiAppName + ":" + streamID + ". ", e);
+            } catch (KeyStoreException e) {
+                throw new SiddhiAppCreationException("Failed to load keystore. Please check " +
+                        "the tls.keystore.type = " + tlsKeystoreType + "  defined in " + siddhiAppName + ":" +
+                        streamID + ". ", e);
+            }
+        }
+    }
+
```
Comment (AnuGayan) [inline component/src/main/java/org/wso2/extension/siddhi/io/hl7/util/Hl7Utils.java:103]: Please remove unwanted new lines

File: component/src/main/java/org/wso2/extension/siddhi/io/hl7/source/RegistrationEventRouting.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+/**
+ * RegistrationEventRouting is used to route messages to the appropriate application.
+ * This class handles all Message Types and Trigger Events.
+ */
+public class RegistrationEventRouting implements ApplicationRouter.AppRoutingData {
+
+    @Override
+    public String getMessageType() {
+
+        return "*";
```
Comment (AnuGayan) [inline component/src/main/java/org/wso2/extension/siddhi/io/hl7/source/RegistrationEventRouting.java:32]: Why we return "*"  here??

File: component/src/main/java/org/wso2/extension/siddhi/io/hl7/util/Hl7Utils.java [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+    public static void doTlsValidation(boolean tlsEnabled, String tlsKeystoreFilepath, String tlsKeystorePassphrase,
+                                String tlsKeystoreType, String siddhiAppName, String streamID) {
+
+        if (tlsEnabled) {
+            try {
+                KeyStore keyStore = KeystoreUtils.loadKeystore(tlsKeystoreFilepath, tlsKeystorePassphrase);
+                KeyStore.getInstance(tlsKeystoreType);
+                KeystoreUtils.validateKeystoreForTlsSending(keyStore);
+            } catch (FileNotFoundException e) {
+                throw new SiddhiAppCreationException("Failed to found the keystore file." +
```
Comment (AnuGayan) [inline component/src/main/java/org/wso2/extension/siddhi/io/hl7/util/Hl7Utils.java]: Shall we change the error message in to "Failed to find .. "

File: README.md [hunk 12]
```diff
@@ -1 +1,44 @@
-# siddhi-io-hl7
\ No newline at end of file
+﻿siddhi-io-hl7
```
Comment (AnuGayan) [inline README.md:1]: Please complete the readme

File: mkdocs.yml [hunk 13]
```diff
@@ -0,0 +1,16 @@
+site_name: Siddhi-IO-Hl7
```
Comment (AnuGayan) [inline mkdocs.yml:1]: Please complete the mkdocs file, (https://github.com/wso2-extensions/siddhi-map-json/blob/master/mkdocs.yml)

File: component/src/test/resources/security/kk.js (hunk 1 above)
Comment (mayuravaani) [reply component/src/test/resources/security/kk.js:1]: It is used to test the invalid file type for tlsEnabled testcases.

File: component/src/main/java/org/wso2/extension/siddhi/io/hl7/source/RegistrationEventRouting.java (hunk 10 above)
Comment (mayuravaani) [reply component/src/main/java/org/wso2/extension/siddhi/io/hl7/source/RegistrationEventRouting.java:32]: Hl7 message has several types. It is mentioned to support all the types.
https://www.hl7.org/documentcenter/public_temp_58BF27F2-1C23-BA17-0CCFAA225A24A762/wg/conf/HL7MSH.htm

**Review by AnuGayan — APPROVED** [review]

**Pull request merged**